From cea9f66912694fc29c0b10ed5102d187eabc672e Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:18 +0100
Subject: [PATCH 18/51] Revert "Duplicate declarations of template default
 argument from rootmap files are just fine."

This reverts commit 4dbec5f492357d012b5de6b2411e8f7b6a7f3e76.
---
 interpreter/llvm-project/clang/lib/Sema/SemaTemplate.cpp | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/interpreter/llvm-project/clang/lib/Sema/SemaTemplate.cpp b/interpreter/llvm-project/clang/lib/Sema/SemaTemplate.cpp
index 850d28ce56..a12a64939c 100644
--- a/interpreter/llvm-project/clang/lib/Sema/SemaTemplate.cpp
+++ b/interpreter/llvm-project/clang/lib/Sema/SemaTemplate.cpp
@@ -3143,14 +3143,6 @@ bool Sema::CheckTemplateParameterList(TemplateParameterList *NewParams,
       Invalid = true;
     }
 
-    if (RedundantDefaultArg &&
-        (((*OldParam)->hasAttr<AnnotateAttr>() &&
-         (*OldParam)->getAttr<AnnotateAttr>()->getAnnotation() == "rootmap") ||
-        ((*NewParam)->hasAttr<AnnotateAttr>() &&
-         (*NewParam)->getAttr<AnnotateAttr>()->getAnnotation() == "rootmap"))) {
-      RedundantDefaultArg = false;
-    }
-
     // [basic.def.odr]/13:
     //     There can be more than one definition of a
     //     ...
-- 
2.43.5

